bot/services/ChatBot.cs

```diff
@@ -44,6 +44,37 @@ public class ChatBot(
 
     public const string StartTimeKey = "http-request-start-time";
 
+    private bool IsAuthorized(ITurnContext<IMessageActivity> turnContext)
+    {
+        // if no tenants are specified, then all are valid
+        if (this.config.VALID_TENANTS.Length == 0)
+        {
+            this.logger.LogDebug("no tenants are specified, all are authorized to use the bot.");
+            return true;
+        }
+
+        // compare the tenant
+        try
+        {
+            string tenantId = turnContext.Activity.ChannelData["tenant"]["id"].ToString();
+            if (this.config.VALID_TENANTS.Contains(tenantId))
+            {
+                this.logger.LogDebug("the tenant {t} is authorized to use the bot.", tenantId);
+                return true;
+            }
+            else

```

Comment:
it may be cleaner to just return false without the else

Reply:
sure, I can do that.